internal/jimm/model.go

```diff
@@ -661,29 +661,35 @@ func (j *JIMM) AddModel(ctx context.Context, user *openfga.User, args *ModelCrea
 
 	mi := builder.JujuModelInfo()
 
-	if err := j.OpenFGAClient.AddControllerModel(
-		ctx,
-		builder.controller.ResourceTag(),
-		builder.model.ResourceTag(),
-	); err != nil {
+	ownerUser := openfga.NewUser(owner, j.OpenFGAClient)
+	modelTag := names.NewModelTag(mi.UUID)
+	controllerTag := builder.controller.ResourceTag()
+
+	return mi, j.addModelPermissions(ctx, ownerUser, modelTag, controllerTag)
```

Comment:
I think the returned error should be wrapped with the `op` const, like the other returns in this method:

```go
if err := j.addModelPermissions(ctx, ownerUser, modelTag, controllerTag); err != nil {
	return nil, errors.E(op, err)
}
```